core/src/main/java/io/fabric8/maven/core/service/openshift/ImageStreamService.java

```diff
@@ -205,29 +209,62 @@ private String findTagSha(OpenShiftClient client, String imageStreamName, String
             if (tags == null || tags.isEmpty()) {
                 continue;
             }
-            // latest tag is the first
+
+            // Iterate all imagestream tags and get the latest one by 'created' attribute
+            TagEvent latestTag = null;
+
             TAG_EVENT_LIST:
             for (NamedTagEventList list : tags) {
                 List<TagEvent> items = list.getItems();
-                if (items == null) {
+                if (items == null || items.isEmpty()) {
                     continue TAG_EVENT_LIST;
                 }
 
-                // latest item is the first
-                for (TagEvent item : items) {
-                    String image = item.getImage();
-                    if (Strings.isNotBlank(image)) {
-                        log.info("Found tag on ImageStream " + imageStreamName + " tag: " + image);
-                        return image;
-                    }
+                for (TagEvent tag : items) {
```

Comment:
Do we know why latest item is no longer the first ? That seems like a behavior that must have changed.

Also, moving to search through all tags to find the latest based on created image do not really solve the issue  does it ? 
another build could be happening for a PR or similar that would create something even later but is not related to this ongoing build - correct ?

Reply:
in other words - using created attribute is better than latest tag now; but ultimately it is as broken as it always been. The code relies on there being zero other builds happening at the same time.

Reply:
in a deterministic flow there should be a passing of a id/label or similar to ensure uniquness.

Reply:
Absolutely missed these comments somehow. (_thats bad_)
Creating one more issue and spinning the discussion there.